tools/run_tests/xds_test_driver/framework/xds_k8s_testcase.py

```diff
@@ -0,0 +1,391 @@
+# Copyright 2020 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import enum
+import hashlib
+import logging
+from typing import Tuple
+
+from absl import flags
+from absl.testing import absltest
+
+from framework import xds_flags
+from framework import xds_k8s_flags
+from framework.infrastructure import k8s
+from framework.infrastructure import gcp
+from framework.infrastructure import traffic_director
+from framework.rpc import grpc_channelz
+from framework.test_app import client_app
+from framework.test_app import server_app
+
+logger = logging.getLogger(__name__)
+flags.adopt_module_key_flags(xds_flags)
+flags.adopt_module_key_flags(xds_k8s_flags)
+
+# Type aliases
+XdsTestServer = server_app.XdsTestServer
+XdsTestClient = client_app.XdsTestClient
+
+
+class XdsKubernetesTestCase(absltest.TestCase):
+    k8s_api_manager: k8s.KubernetesApiManager
+    gcp_api_manager: gcp.api.GcpApiManager
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def setUpClass(cls):
+        # GCP
+        cls.project: str = xds_flags.PROJECT.value
+        cls.network: str = xds_flags.NETWORK.value
+        cls.gcp_service_account: str = xds_k8s_flags.GCP_SERVICE_ACCOUNT.value
+        cls.td_bootstrap_image = xds_k8s_flags.TD_BOOTSTRAP_IMAGE.value
+
+        # Base namespace
+        # todo(sergiitk): generate for each test
+        cls.namespace: str = xds_flags.NAMESPACE.value
+
+        # Test server
+        cls.server_image = xds_k8s_flags.SERVER_IMAGE.value
+        cls.server_name = xds_flags.SERVER_NAME.value
+        cls.server_port = xds_flags.SERVER_PORT.value
+        cls.server_xds_host = xds_flags.SERVER_NAME.value
+        cls.server_xds_port = xds_flags.SERVER_XDS_PORT.value
+
+        # Test client
+        cls.client_image = xds_k8s_flags.CLIENT_IMAGE.value
+        cls.client_name = xds_flags.CLIENT_NAME.value
+        cls.client_port = xds_flags.CLIENT_PORT.value
+        cls.client_port_forwarding = xds_k8s_flags.CLIENT_PORT_FORWARDING.value
+
+        # Resource managers
+        cls.k8s_api_manager = k8s.KubernetesApiManager(
+            xds_k8s_flags.KUBE_CONTEXT.value)
+        cls.gcp_api_manager = gcp.api.GcpApiManager()
+
+    def setUp(self):
+        # todo(sergiitk): generate for each test
+        self.server_namespace = self.namespace
+        self.client_namespace = self.namespace
+
+        # Init this in child class
+        self.server_runner = None
```

Comment:
Maybe a Java centric approach, but this would seem less error-prone to define `getServer/ClientRunner` methods instead, and have the child class override those methods.